packages/studio/src/components/IFramePortal.tsx

```diff
@@ -1,27 +1,50 @@
 import { createPortal } from "react-dom";
-import { Dispatch, PropsWithChildren, SetStateAction, useEffect } from "react";
+import {
+  Dispatch,
+  PropsWithChildren,
+  SetStateAction,
+  useEffect,
+  useRef,
+} from "react";
+import useStudioStore from "../store/useStudioStore";
+import { EXPANDED_VIEWPORTS } from "./Viewport/defaults";
+import { twMerge } from "tailwind-merge";
 
 export default function IFramePortal(
   props: PropsWithChildren<{
-    className?: string;
     title: string;
     iframeEl: HTMLIFrameElement | null;
     setIframeEl: Dispatch<SetStateAction<HTMLIFrameElement | null>>;
   }>
 ) {
+  const previewRef = useRef<HTMLDivElement>(null);
   const iframeDocument = props.iframeEl?.contentWindow?.document;
+  const [viewportDimensions] = useStudioStore((store) => [
+    store.pagePreview.viewportDimensions,
+  ]);
+  const dimensions =
+    viewportDimensions.width * (previewRef.current?.clientHeight ?? 0) >
+    viewportDimensions.height * (previewRef.current?.clientWidth ?? 0)
+      ? " w-full "
+      : " h-full ";
+  const id = viewportDimensions.name.replace(/\s+/g, "").toLowerCase();
+  const iframeCSS = twMerge(
+    "mr-auto ml-auto border-2",
+    EXPANDED_VIEWPORTS[id].css,
```

Comment:
What do you think of also storing the current viewport's CSS in `store.pagePreview`? Then, we wouldn't need to do this lookup.

Reply:
Hmmm I'm not too sure. It doesn't make total sense to me rn, right now it's calling the newly clicked viewport's css (which I hardcoded in Viewport/defaults). I could hardcode them into the store.pagePreview instead? As a key: value graph...?